src/app/router/Welcome/Welcome.tsx

```diff
@@ -17,34 +18,36 @@ const connectorRoutes = getConnectorRoutes();
 function Welcome() {
   return (
     <SettingsProvider>
-      <Router>
-        <Toaster />
-        <Routes>
-          <Route path="/" element={<Layout />}>
-            <Route index element={<SetPassword />} />
-            <Route path="choose-path">
-              <Route index={true} element={<ChooseConnectorPath />}></Route>
-              <Route path="choose-connector">
-                <Route
-                  index={true}
-                  element={
-                    <ChooseConnector
-                      title={i18n.t("translation:choose_connector.title")}
-                      description={i18n.t(
-                        "translation:choose_connector.description"
-                      )}
-                      connectorRoutes={connectorRoutes}
-                    />
-                  }
-                ></Route>
-                {renderRoutes(connectorRoutes)}
+      <AccountProvider>
```

Comment:
why do we need the AccountProvider here?

Reply:
cause AccountContext has all the logic to handle this. we use AccountProvider in options router as well to handle the such case

Reply:
https://github.com/getAlby/lightning-browser-extension/pull/2928/files#diff-4cbea9de97e0df6000b0839d65825e3f81f7a5cdf85eed529d3adeadaba37ee0R143

Reply:
but that didn't handled the case when extension is locked and user wants to access any route. made little corrections in accountProvider.

Reply:
now we support both. 
1. if extension is locked and user tries to access the screen - redirect to options page. now options router has security checks  so it will redirect to unlock page by default

2. if extension is unlocked and user tried to access welcome page. we redirec to options.html/wallet - as now we are wrapping welcome router under accountProvider. and provider has that checks

so it solves both